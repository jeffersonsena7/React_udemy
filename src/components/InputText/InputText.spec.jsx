import { render, screen } from '@testing-library/react';
import { InputText } from '.';
import userEvent from '@testing-library/user-event';

describe('<InputText />', () => {
  //
  it('should have a value of searchValue', () => {
    const fn = jest.fn();
    render(<InputText handleChange={fn} searchValue={'testando'} />);
    expect(screen.getByPlaceholderText(/type you search/i).value).toBe('testando');
  });
  //deve chamar a função handleChange em cada tecla pressionada
  it('should call handleChange function on each key pressed', () => {
    const fn = jest.fn();
    render(<InputText handleChange={fn} searchValue="o valor" />);

    const input = screen.getByPlaceholderText(/type you search/i);
    const value = 'o valor';

    userEvent.type(input, value);

    expect(input.value).toBe(value);
    expect(fn).toHaveBeenCalledTimes(value.length);
  });

  //agora fazer um snapshot, caso mude alguma coisa do código apresenta erro no teste
  it('should match snapshot', () => {
    const fn = jest.fn();
    const { container } = render(<InputText handleChange={fn} searchValue="" />);
    expect(container).toMatchSnapshot();
  });
});
