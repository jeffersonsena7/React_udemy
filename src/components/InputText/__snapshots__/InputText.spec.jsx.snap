// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputText /> should match snapshot 1`] = `
<div>
  <input
    class="input-text"
    placeholder="type you search"
    type="search"
    value=""
  />
</div>
`;
